src/business-logic/services/rundown-timeline-service.ts

```diff
@@ -224,4 +224,17 @@ export class RundownTimelineService implements RundownService {
 
     await this.rundownRepository.saveRundown(rundown)
   }
+
+  public async replacePiece(rundownId: string, pieceToBeReplaced: Piece, newPiece: Piece): Promise<void> {
```

Comment:
Do you have any suggestions?

Reply:
The best I have currently is `replacePieceInOnAirOrNextPart`.

Reply:
> replacePiece does not sound like it only supports replacing on the on air and next part.

That's because it supports replacing a Piece on any Part ;)
The Piece is replaced on the Part that `pieceToBeReplaced` belongs to. Doesn't matter if that is OnAir, the next one or a third.

Reply:
So I think the current naming is appropriate

Reply:
I stand corrected.
I have updated the name.